spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializer.java

```diff
@@ -40,7 +40,12 @@
 @JsonComponent
 public class MetricJsonSerializer {
 
-	final static DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+	final static ThreadLocal<DateFormat> df = new ThreadLocal<DateFormat>(){
+			@Override
+			protected DateFormat initialValue() {
+					return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+			}
+	};
```

Comment:
I rather not let it to be configurable as it's a bit of hassle downstream. We have set it as the default format for timestamp, and allowing users to change it could break downstream apps such as the collector. Timestamp representation in json can either be millis since epoch or ISO 8601 which is the format currently being used. Consumers of the payload can always apply required transformations if say they want to store in a numerical format in a database for instance.